lib/Parser/Parse.cpp

```diff
@@ -10925,7 +10925,8 @@ ParseNodePtr Parser::Parse(LPCUTF8 pszSrc, size_t offset, size_t length, charcou
         }
     }
 
-    if (isModuleSource && !isDeferred)
+    // It's possible for the module global to be defer-parsed in debug scenarios.
+    if (isModuleSource && (!isDeferred || (isDeferred && grfscr & fscrGlobalCode)))
```

Comment:
We hit this when we try to do source rundown and reparse for debug attach scenario. I think it might be a real scenario if user opens F12. Ordinary global functions work fine in that case but the module code needs to generate the function body wrapper or else import/export statements will be treated as syntax error.